### ISSUE TITLE
NPE in Mutation when deletions is null

### ISSUE THREAD

**Issue by KStubbings** [thinkaurelius/titan#40]
NPE is thrown in Mutation when deletions is null, however deletions is always null when mutateIndex is called from addIndexEntry.  

Stack trance:

   java.lang.NullPointerException
    at java.util.ArrayList.addAll(Unknown Source)
    at com.thinkaurelius.titan.diskstorage.writeaggregation.Mutation.merge(Mutation.java:44)
    at com.thinkaurelius.titan.diskstorage.writeaggregation.BufferStoreMutator.mutate(BufferStoreMutator.java:59)
    at com.thinkaurelius.titan.diskstorage.writeaggregation.BufferStoreMutator.mutateIndex(BufferStoreMutator.java:45)
    at com.thinkaurelius.titan.graphdb.database.StandardTitanGraph.addIndexEntry(StandardTitanGraph.java:906)
    at com.thinkaurelius.titan.graphdb.database.StandardTitanGraph.persist(StandardTitanGraph.java:696)2012-07-19 12:23:26,768-INFO -AbstractFormatter    OBJY / IG Error

```
at com.thinkaurelius.titan.graphdb.database.StandardTitanGraph.save(StandardTitanGraph.java:642)
at com.thinkaurelius.titan.graphdb.transaction.StandardPersistTitanTx.commit(StandardPersistTitanTx.java:171)
at com.thinkaurelius.titan.graphdb.blueprints.TitanBlueprintsTransaction.stopTransaction(TitanBlueprintsTransaction.java:30)
```

**Comment by dalaro** [comment]
Thanks for the report.

This NPE was triggering whenever a single transaction set p=v on one vertex while removing it from some other vertex, where p is an indexed but non-unique property and v is some value for p.  I wrote a test case for this scenario that generated your stack trace snippet and I fixed the NPE by adding null checks in diskstorage.writeaggregation.Mutation's merge method.  I'm planning to cherry-pick these from titan0.1 into the master.  I think this should address the problem.

Although I don't think I've missed additional scenarios that cause this particular NPE, I wonder what we would find if we ran a static analysis package on Titan core...
